File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];

        for (int i = 0; i < n; i++) {
            int combined = arr1[i] | arr2[i];

            String binary = Integer.toBinaryString(combined);

            while (binary.length() < n) {
                binary = "0" + binary;
            }

            binary = binary.replace('1', '#').replace('0', ' ');

            answer[i] = binary;
        }
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;

    for (int i = 0; i < n; i++) {
        int combined = arr1[i] | arr2[i]; 
        string row = "";

        for (int j = n - 1; j >= 0; j--) {
            if ((combined >> j) & 1) {
                row += "#";
            } else {
                row += " ";
            }
        }
        answer.push_back(row);
    }

    return answer;
}